src/runtime-rs/crates/resource/src/share_fs/virtio_fs_share_mount.rs

```diff
@@ -224,4 +225,18 @@ impl ShareFsMount for VirtiofsShareMount {
 
         Ok(())
     }
+
+    async fn cleanup_hostpath(&self, sid: &str) -> Result<()> {
+        // Unmount ro path
+        let host_ro_dest = get_host_ro_shared_path(sid);
+        umount_all(host_ro_dest.clone(), true).context("failed to umount ro path")?;
+        fs::remove_dir_all(host_ro_dest).context("failed to remove ro path")?;
+        // As the rootfs and volume have been umounted before calling this function, so just remove the rw dir directly
+        let host_rw_dest = get_host_rw_shared_path(sid);
+        fs::remove_dir_all(host_rw_dest).context("failed to remove rw path")?;
+        // remove the host share directory
+        let host_path = get_host_shared_path(sid);
+        fs::remove_dir_all(host_path).context("failed to remove host shared path")?;
```

Comment:
I am curious why we need to remove dir three times? Seems only remove host_path is ok, by this, host_ro_dest and host_rw_dest will also be removed.

Reply:
I think this is because when cleaning up the host share path, there are multiple directories to be handled, which is step by step. And later, we are going to support sandbox bind mount as well, so removing each directory when it is ready to delete makes more sense here.

Reply:
ok